lib/decoder/real-time.js

```diff
@@ -0,0 +1,66 @@
+const url = require('url')
+
+// https://github.com/PRX/Infrastructure/blob/master/cdn/dovetail3-cdn.yml#L289
+const FIELDS = [
+  'timestamp',
+  'c-ip',
+  'sc-status',
+  'cs-method',
+  'cs-uri-stem',
+  'cs-user-agent',
+  'cs-referer',
+  'x-forwarded-for',
+  'sc-content-len',
+  'sc-range-start',
+  'sc-range-end',
+]
+const ASCII = /^[\x00-\xFF]*$/
+
+/**
+ * check for cloudfront real-time log events, with known TSV fields
+ */
+exports.detect = function (buffer) {
+  const str = buffer.toString('utf-8')
+  return ASCII.test(str) && str.split('\t').length === FIELDS.length
+}
+
+/**
+ * parse tab separated log lines
+ */
+exports.decode = async function (buffer) {
+  const parts = buffer.toString('utf-8').trim().split('\t')
+  const data = {}
+  FIELDS.forEach((name, index) => {
+    if (parts[index] === '-') {
+      data[name] = null
+    } else {
+      data[name] = parts[index]
+    }
+  })
+
+  // only 2XX GET requests count
+  if (data['cs-method'] !== 'GET' || data['sc-status'] < '200' || data['sc-status'] >= '300') {
```

Comment:
GT/LT comparisons on strings feels a little wonky to me, even though I know it works.

Reply:
+1 will change to a regex.  I'm hesitant to parseInt them, as I guess there are some special "leading 0s" values they'll put in there, like `000`?  But `.match(/2[0-9][0-9]/)` should do it.